src/ring_theory/ideals.lean

```diff
@@ -13,11 +13,139 @@ open set function
 
 open_locale classical big_operators
 
+/-- Ideal in a commutative ring is an additive subgroup `s` such that
+`a * b ∈ s` whenever `b ∈ s`. -/
+structure ideal (α : Type u) [ring α] extends add_subgroup α :=
+(mul_mem_left : ∀ {a b}, b ∈ carrier → a * b ∈ carrier)
+(mul_mem_right : ∀ {a b}, a ∈ carrier → a * b ∈ carrier)
+
+namespace ideal
+variables [comm_ring α] (I : ideal α) (J : ideal α)
+
+-- these are all copied from submodule
+instance : has_coe (ideal α) (set α) := ⟨λ s, s.carrier⟩
```

Comment:
Submodule used `has_coe_t`, [not sure what the difference is](https://leanprover.zulipchat.com/#narrow/stream/113489-new-members/topic/has_coe_t.20vs.20has_coe)